test/unit/client-daemonset.bats

```diff
@@ -401,28 +397,26 @@ load _helpers
 @test "client/DaemonSet: gossip encryption disabled in client DaemonSet by default" {
   cd `chart_dir`
   local actual=$(helm template \
-      -x templates/client-daemonset.yaml  \
+      -s templates/client-daemonset.yaml  \
       . | tee /dev/stderr |
       yq '.spec.template.spec.containers[] | select(.name=="consul") | .env[] | select(.name == "GOSSIP_KEY") | length > 0' | tee /dev/stderr)
   [ "${actual}" = "" ]
 }
 
 @test "client/DaemonSet: gossip encryption disabled in client DaemonSet when clients are disabled" {
```

Comment:
This test is confusing to me. It seems a bit redundant given that every feature is disabled when clients are disabled, and I don't think we test this for every feature. I'd delete it, but up to you!

Reply:
agreed!